eng/resolveContract.targets

```diff
@@ -60,11 +65,36 @@
                                       ReferenceAssembly="" />
     </ItemGroup>
   </Target>
-  
+
+  <!-- Generate a .txt file with all public types of the project referenced by ProjectForGenAPIDocIdGeneration (e.g. System.Private.CoreLib or System.Private.Uri implementation assembly).
+       This file is then later on passed to GenAPI as the list of types which should be "split out" with the conditional compilation logic. -->
+  <Target Name="GenerateConditionalTypeList"
+          Condition="'@(ProjectReference)' != '' and @(ProjectReference->AnyHaveMetadataValue('Identity', '$(ProjectForGenAPIDocIdGeneration)')) and ('$(ContractTypesPartiallyMoved)' == 'true' or '$(ContractTypesPartiallyMoved)' == 'true')">
+    <PropertyGroup>
+      <!-- generate conditional compilation symbols to hide types implemented in contracts so we can include the .cs in the System.Private.CoreLib ref -->
+      <GenAPIConditionalTypeList>$(IntermediateOutputPath)GenAPIConditionalTypeList.txt</GenAPIConditionalTypeList>
+      <GenAPIConditionalTypeListSymbol Condition="'$(GenAPIConditionalTypeListSymbol)' == ''">!HIDE_CONTRACT_TYPES</GenAPIConditionalTypeListSymbol>
```

Comment:
Have folks discussed the naming here?  This is confusing to me.  I'd prefer something like `BUILDING_REFERENCE` or something.  @stephentoub @tannergooding what is your preference since you're likely to be seeing this the most?

Reply:
With the changes from https://github.com/dotnet/arcade/pull/10003 this was easier to do so I switched to using a single `BUILDING_CORELIB_REFERENCE` symbol.